### PR TITLE
Aad application: App Roles, Key Credentials, Password Credentials, Authentication Behaviours, PreAuthorizationApplications

### PR DESCRIPTION
**Pull request by kasaxena5**
#### Pull Request (PR) description
This PR adds the following complex cim instances to AAD Application:
App Roles
Key Credentials (Added Custom Logic to manage addition and removal of certificates)
Password Credentials (Read Only Property)
Authentication Behaviours (This parameter is available only in Beta cmdlet, added custom logic to fetch from beta cmdlet)
PreAuthorizationApplications

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADApplication/MSFT_AADApplication.psm1 [hunk 1]
```diff
@@ -163,6 +183,110 @@ function Get-TargetResource
         else
         {
             Write-Verbose -Message 'An instance of Azure AD App was retrieved.'
+
+
+            $AADBetaApp= Get-MgBetaApplication -Property "id,displayName,appId,authenticationBehaviors" -ApplicationId $ObjectID -ErrorAction SilentlyContinue
+            $AADAppKeyCredentials = Get-MgApplication -Property "keyCredentials" -ApplicationId $ObjectID -ErrorAction SilentlyContinue
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADApplication/MSFT_AADApplication.psm1:189]: Isn't this already retrieved by line 171 above? Do we need to make yet another call across the wire? Can't we just do one single call to the beta call for the entire Get-TargetResource function?

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Please make sure you add an entry to the change log.

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADApplication/MSFT_AADApplication.psm1 (hunk 1 above)
Comment (kasaxena5) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADApplication/MSFT_AADApplication.psm1:189]: I am not sure if there are any changes in the format or supportability of other properties in beta cmdlet. So I chose to use beta to get only authenticationBehaviors property

**Comment by kasaxena5** [comment]
> Please make sure you add an entry to the change log.

Added

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**